### PR TITLE
(HP-1388) add repository info to data_repositories

### PR DESCRIPTION
**Pull request by george42-ctds**
Jira Ticket: [HP-1388](https://ctds-planx.atlassian.net/browse/HP-1388)

### Improvements

* Populate both the Gen3 tag and the data_repositories when only a repository name is indicated during study registration

### Deployment changes

[HP-1388]: https://ctds-planx.atlassian.net/browse/HP-1388?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/StudyRegistration/utils.tsx [hunk 1]
```diff
@@ -35,13 +35,24 @@ export const preprocessStudyRegistrationMetadata = async (username, metadataID,
     metadataToUpdate[STUDY_DATA_FIELD][studyRegistrationTrackingField] = username;
 
     // add all repository_study_ids as separate objects
+    var tempStudyIDObj = [{}]
```
Comment (mfshao) [inline src/StudyRegistration/utils.tsx:38]:
```suggestion
    let tempStudyIDObj = [];
```

File: src/StudyRegistration/utils.tsx [hunk 2]
```diff
@@ -35,13 +35,24 @@ export const preprocessStudyRegistrationMetadata = async (username, metadataID,
...
     if (updatedValues.repository_study_ids?.length > 0) {
-      const tempStudyIDObj = updatedValues.repository_study_ids.map((studyId) => ({
+      tempStudyIDObj = updatedValues.repository_study_ids.map((studyId) => ({
         repository_name: updatedValues.repository,
         repository_study_ID: studyId,
       }));
-      metadataToUpdate[STUDY_DATA_FIELD].study_metadata.metadata_location.data_repositories = tempStudyIDObj;
+    } else {
+      if (updatedValues.repository) {
```
Comment (mfshao) [inline src/StudyRegistration/utils.tsx:45]: these 2 lines can be 1 single line of `else if (...)`

File: src/StudyRegistration/utils.tsx [hunk 3]
```diff
@@ -35,13 +35,24 @@ export const preprocessStudyRegistrationMetadata = async (username, metadataID,
...
+        tempStudyIDObj = [{
+          repository_name: updatedValues.repository,
+          repository_study_ID: "",
+          repository_study_link: "",
+          repository_persistent_ID: ""
```
Comment (mfshao) [inline src/StudyRegistration/utils.tsx:50]: minor: we prefer using single quotes for strings

File: src/StudyRegistration/utils.tsx (hunk 1 above)
Comment (george42-ctds) [reply src/StudyRegistration/utils.tsx]: Omitting the curly braces is giving a type error on the new `tempStudyIDObj`. Retaining the curly braces makes the error go away.

**Review by mfshao — APPROVED** [review]

**Pull request merged**